sky/provision/fluidstack/instance.py

```diff
@@ -0,0 +1,271 @@
+            if (isinstance(e, utils.FluidstackAPIError) and
+                    'Machine is already terminated' in str(e)):
+                logger.debug(f'Instance {inst_id} is already terminated.')
+                continue
+            with ux_utils.print_exception_no_traceback():
+                raise RuntimeError(
+                    f'Failed to terminate instance {inst_id}: '
+                    f'{common_utils.format_exception(e, use_bracket=False)}'
+                ) from e
+
+
+def get_cluster_info(
+        region: str,
+        cluster_name_on_cloud: str,
+        provider_config: Optional[Dict[str, Any]] = None) -> common.ClusterInfo:
+    del region, provider_config  # unused
+    running_instances = _filter_instances(cluster_name_on_cloud, ['running'])
+    instances: Dict[str, List[common.InstanceInfo]] = {}
+
+    subprocess_utils.run_in_parallel(get_internal_ip,
+                                     list(running_instances.values()))
+    head_instance_id = None
+    for instance_id, instance_info in running_instances.items():
+        instance_id = instance_info['id']
+        instances[instance_id] = [
+            common.InstanceInfo(
+                instance_id=instance_id,
+                internal_ip=instance_info['internal_ip'],
+                external_ip=instance_info['ip_address'],
+                ssh_port=instance_info['ssh_port'],
+                tags={},
+            )
+        ]
+        if instance_info['hostname'].endswith('-head'):
+            head_instance_id = instance_id
+
+    return common.ClusterInfo(
+        instances=instances,
+        head_instance_id=head_instance_id,
+    )
+
+
+def query_instances(
+    cluster_name_on_cloud: str,
+    provider_config: Optional[Dict[str, Any]] = None,
+    non_terminated_only: bool = True,
+) -> Dict[str, Optional[status_lib.ClusterStatus]]:
+    """See sky/provision/__init__.py"""
+    assert provider_config is not None, (cluster_name_on_cloud, provider_config)
+    instances = _filter_instances(cluster_name_on_cloud, None)
+    instances = _filter_instances(cluster_name_on_cloud, None)
+    status_map = {
+        'provisioning': status_lib.ClusterStatus.INIT,
+        'requesting': status_lib.ClusterStatus.INIT,
+        'create': status_lib.ClusterStatus.INIT,
+        'customizing': status_lib.ClusterStatus.INIT,
+        'stopping': status_lib.ClusterStatus.STOPPED,
+        'stop': status_lib.ClusterStatus.STOPPED,
+        'start': status_lib.ClusterStatus.INIT,
+        'reboot': status_lib.ClusterStatus.STOPPED,
+        'rebooting': status_lib.ClusterStatus.STOPPED,
+        'stopped': status_lib.ClusterStatus.STOPPED,
+        'starting': status_lib.ClusterStatus.INIT,
+        'running': status_lib.ClusterStatus.UP,
+        'failed to create': status_lib.ClusterStatus.INIT,
+        'timeout error': status_lib.ClusterStatus.INIT,
+        'out of stock': status_lib.ClusterStatus.INIT,
+    }
```

Comment:
How about we explicitly list out the terminating/terminated state in the map and map them to `None`?